app/uk/gov/hmrc/gform/sharedmodel/formtemplate/ComponentType.scala

```diff
@@ -26,7 +26,11 @@ import scala.collection.immutable._
 
 sealed trait ComponentType
 
-case class Text(constraint: TextConstraint, value: Expr, displayWidth: DisplayWidth = DisplayWidth.DEFAULT)
+case class Text(
+  constraint: TextConstraint,
+  value: Expr,
+  displayWidth: DisplayWidth = DisplayWidth.DEFAULT,
+  roundingMode: RoundingMode = RoundingMode.defaultRoundingMode)
```

Comment:
Why is `roundingMode ` here? 

Text should have no knowledge of rounding. That knowledge should be responsibility of constraint, namely `Number` etc. And as I can see you've also added it there.